### PR TITLE
add odf installation verification

### PR DESCRIPTION
**Pull request by ebattat**
## Type of change
Note: Fill **x** in []
- [ ] bug
- [x] enhancement
- [ ] documentation
- [ ] dependencies

## Description

Add ODF installation verification by using the rook-ceph-tools pod's 'ceph check' command. 
Return 'HEALTH_OK' if all is working properly and raise error ODFInstallationFailed is case of failure

### CODE REVIEW

File: benchmark_runner/common/oc/oc.py [hunk 1]
```diff
@@ -244,6 +244,19 @@ def wait_for_dv_status(self,
...
+        :return: True ODF passed, False failed
+        """
+        self.run(""" oc patch ocsinitialization ocsinit -n openshift-storage --type json --patch '[{ "op": "replace", "path": "/spec/enableCephTools", "value": true }]' """)
+        rook_ceph_tools_pod = self._get_pod_name(pod_name='rook-ceph-tools', namespace='openshift-storage')
+        result = self.run(f"oc -n openshift-storage rsh {rook_ceph_tools_pod} ceph health")
+        if 'HEALTH_OK' == result.strip():
+            return True
+        else:
+            return False
+
```
Comment (RobertKrawitz) [inline benchmark_runner/common/oc/oc.py:259]: Minor:

return 'HEALTH_OK' == result.strip()

File: benchmark_runner/common/ocp_resources/create_odf.py [hunk 2]
```diff
@@ -69,4 +69,4 @@ def create_odf(self):
                     self.wait_for_ocp_resource_create(resource='odf',
                                                       verify_cmd='oc get pod -n openshift-storage | grep osd | grep -v prepare | wc -l',
                                                       count_openshift_storage=True)
-        return True
+        return ODFInstallationFailed() if not self.__oc.verify_odf_installation() else self.__oc.verify_odf_installation()
```
Comment (RobertKrawitz) [inline benchmark_runner/common/ocp_resources/create_odf.py:72]: Raise the exception, don't return it.

**Review by RobertKrawitz — CHANGES_REQUESTED** [review]

File: benchmark_runner/common/ocp_resources/create_odf.py (hunk 2 above)
Comment (ebattat) [reply benchmark_runner/common/ocp_resources/create_odf.py]: Good catch, fix it

File: benchmark_runner/common/oc/oc.py (hunk 1 above)
Comment (ebattat) [reply benchmark_runner/common/oc/oc.py]: Fixed

File: benchmark_runner/common/oc/oc.py [hunk 3]
```diff
@@ -244,6 +244,16 @@ def wait_for_dv_status(self,
             current_wait_time += OC.SLEEP_TIME
         raise DVStatusTimeout(status=status)
 
+    def verify_odf_installation(self):
+        """
+        This method verifies ODF installation
+        :return: True ODF passed, False failed
+        """
+        self.run(""" oc patch ocsinitialization ocsinit -n openshift-storage --type json --patch '[{ "op": "replace", "path": "/spec/enableCephTools", "value": true }]' """)
```
Comment (malayparida2000) [inline benchmark_runner/common/oc/oc.py:252]: Please use the storagecluster to enable the toolbox, We are planning to remove using the ocsinitialization cr for enabling the toolbox in 4.15 or some future release.
so you should replace your command to
oc patch storagecluster ocs-storagecluster -n openshift-storage --type json --patch '[{ "op": "replace", "path": "/spec/enableCephTools", "value": true }]'

**Review by malayparida2000 — CHANGES_REQUESTED** [review]

File: benchmark_runner/common/oc/oc.py (hunk 3 above)
Comment (ebattat) [reply benchmark_runner/common/oc/oc.py]: Thx for your comment.
I have updated it with new command.
Pls let me know if it ok.

Comment (malayparida2000) [reply benchmark_runner/common/oc/oc.py]: Yeah now the patch command looks good to me now.

**Review by RobertKrawitz — APPROVED** [review]

**Pull request merged**